SPECS/xapi-storage-script.spec

```diff
@@ -17,73 +18,56 @@ BuildRequires: message-switch-devel
+%autosetup
 
 %build
 make
 mv main.native xapi-storage-script
-./xapi-storage-script --help=groff > xapi-storage-script.1
-sed -e "s|@LIBEXECDIR@|%{_libexecdir}|g" xapi-storage-script-conf.in > xapi-storage-script.conf
+./xapi-storage-script --help=groff > xapi-storage-script.8 && gzip xapi-storage-script.8
+sed -e "s|@LIBEXECDIR@|%{_libexecdir}|g" %{SOURCE3} > xapi-storage-script.conf
 
 %install
- 
-mkdir -p %{buildroot}/%{_sbindir}
-install -m 0755 xapi-storage-script %{buildroot}/%{_sbindir}/xapi-storage-script
-mkdir -p %{buildroot}%{_sysconfdir}/init.d
-install -m 0755 xapi-storage-script-init %{buildroot}%{_sysconfdir}/init.d/xapi-storage-script
-mkdir -p %{buildroot}/%{_libexecdir}/xapi-storage-script
-mkdir -p %{buildroot}/%{_libexecdir}/xapi-storage-script/volume
-mkdir -p %{buildroot}/%{_libexecdir}/xapi-storage-script/datapath
-mkdir -p %{buildroot}/etc
-install -m 0644 xapi-storage-script.conf %{buildroot}/etc/xapi-storage-script.conf
-mkdir -p %{buildroot}%{_mandir}/man2
-install -m 0644 xapi-storage-script.1 %{buildroot}%{_mandir}/man2/xapi-storage-script.1
-gzip %{buildroot}%{_mandir}/man2/xapi-storage-script.1
+mkdir -p %{buildroot}%{_libexecdir}/xapi-storage-script
+mkdir -p %{buildroot}%{_libexecdir}/xapi-storage-script/volume
+mkdir -p %{buildroot}%{_libexecdir}/xapi-storage-script/datapath
+%{__install} -D -m 0755 xapi-storage-script %{buildroot}%{_sbindir}/xapi-storage-script
+%{__install} -D -m 0644 xapi-storage-script.conf %{buildroot}%{_sysconfdir}/xapi-storage-script.conf
+%{__install} -D -m 0644 xapi-storage-script.8.gz %{buildroot}%{_mandir}/man8/xapi-storage-script.1.gz
+%{__install} -D -m 0644 %{SOURCE1} %{buildroot}%{_unitdir}/xapi-storage-script.service
+%{__install} -D -m 0644 %{SOURCE2} %{buildroot}%{_sysconfdir}/sysconfig/xapi-storage-script
 
 %post
-case $1 in
-  1) # install
-    /sbin/chkconfig --add xapi-storage-script
-    ;;
-  2) # upgrade
-    /sbin/chkconfig --del xapi-storage-script
-    /sbin/chkconfig --add xapi-storage-script
-    ;;
-esac
+%systemd_post xapi-storage-script.service
 
 %preun
-case $1 in
-  0) # uninstall
-    /sbin/service xapi-storage-script stop >/dev/null 2>&1 || :
-    /sbin/chkconfig --del xapi-storage-script
-    ;;
-  1) # upgrade
-    ;;
-esac
+%systemd_preun xapi-storage-script.service
+
+%postun
+%systemd_postun_with_restart xapi-storage-script.service
 
 %files
-%{_sbindir}/xapi-storage-script
-/etc/init.d/xapi-storage-script
-%config(noreplace) /etc/xapi-storage-script.conf
 %{_libexecdir}/xapi-storage-script
 %{_libexecdir}/xapi-storage-script/volume
 %{_libexecdir}/xapi-storage-script/datapath
-%{_mandir}/man2/xapi-storage-script.1.gz
+%{_sbindir}/xapi-storage-script
+%{_mandir}/man8/xapi-storage-script.1.gz
+%{_unitdir}/xapi-storage-script.service
+%config(noreplace) %{_sysconfdir}/sysconfig/xapi-storage-script
+%config(noreplace) %{_sysconfdir}/xapi-storage-script.conf
 
 %changelog
-* Mon May 16 2016 Si Beaumont <simon.beaumont@citrix.com> - 0.12.1-2
```

Comment:
This is still a valid changelog. You should add your and bump the release number to 3.